pincer/objects/guild/guild.py

```diff
@@ -486,6 +496,260 @@ async def kick(self, member_id: int):
+        """
+        data = await self._http.patch(
+            f"guilds/{self.id}/roles",
+            data={"id": id, "position": position},
+            headers={"X-Audit-Log-Reason": reason}
+            if reason is not None
+            else {}
+        )
+        for i in data:
+            yield Role.from_dict(construct_client_dict(self._client, i))
+
+    @overload
+    async def edit_role(
+        self,
+        id: Snowflake,
+        reason: Optional[str] = None,
+        *,
+        name: Optional[str] = None,
+        permissions: Optional[str] = None,
+        color: Optional[int] = None,
+        hoist: Optional[bool] = None,
+        icon: Optional[str] = None,
+        unicode_emoji: Optional[str] = None,
+        mentionable: Optional[bool] = None,
+    ) -> Role:
+        """|coro|
+        Edits a role.
+        Requires the `MANAGE_ROLES` permission.
+
+        Parameters
+        ----------
+        id : :class:`~pincer.utils.snowflake.Snowflake`
+            The role ID
+        reason : Optional[:class:`str`]
+            Reason for editing the role |default| :data:`None`
+        name : Optional[:class:`str`]
+            Name of the role |default| :data:`None`
+        permissions : Optional[:class:`str`]
+            Bitwise value of the enabled/disabled
+            permissions |default| :data:`None`
+        color : Optional[:class:`int`]
+            RGB color value |default| :data:`None`
+        hoist : Optional[:class:`bool`]
+            Whether the role should be displayed
+            separately in the sidebar |default| :data:`None`
+        icon : Optional[:class:`str`]
+            The role's icon image (if the guild has
+            the `ROLE_ICONS` feature) |default| :data:`None`
+        unicode_emoji : Optional[:class:`str`]
+            The role's unicode emoji as a standard emoji (if the guild
+            has the `ROLE_ICONS` feature) |default| :data:`None`
+        mentionable : Optional[:class:`bool`]
+            Whether the role should be mentionable |default| :data:`None`
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.role.Role`
+            The edited role object.
+        """
+        ...
+
+    async def edit_role(
+        self,
+        id: Snowflake,
+        reason: Optional[str] = None,
+        **kwargs
+    ) -> Role:
+        return Role.from_dict(
+            construct_client_dict(
+                self._client,
+                await self._http.patch(
+                    f"guilds/{self.id}/roles/{id}",
+                    data=kwargs,
+                    headers={"X-Audit-Log-Reason": reason}
+                    if reason is not None
+                    else {},
+                ),
+            )
+        )
+
+    async def delete_role(self, id: Snowflake, reason: Optional[str] = None):
+        """|coro|
+        Deletes a role.
+        Requires the `MANAGE_ROLES` permission.
+        Returns `204 No Content` on success.
```

Comment:
remove this line. it can confuse people to believe that the function returns something